### PR TITLE
DrivAerML: Lower LR sweep (2e-4, 2.5e-4, 3e-4) at T_max=30 with 4L/512d

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The DrivAerML baseline lr=5e-4 was set without a thorough LR sweep. For a complex 3D geometry with high surface point count (50k points per sample), lower learning rates often converge to better minima by taking smaller steps in the loss landscape. We sweep lr ∈ {2e-4, 2.5e-4, 3e-4} at T_max=30 to test whether the baseline LR is already near-optimal or if lower values help. This is a focused LR sensitivity analysis keeping everything else fixed at the 4L/512d golden config.

## Instructions
Test three lower LRs at T_max=30. Keep 4L/512d/8H architecture, AdamW, gc=1.0, WD=1e-2.

**Run 1 (CUDA 0) — lr=2e-4 + T_max=30:**
```bash
CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --optimizer adamw \
  --lr 2e-4 \
  --weight-decay 1e-2 \
  --grad-clip 1.0 \
  --cosine-t-max 30 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb-group drivaerml-lr-sweep \
  --wandb-name driv-lr2e4-Tmax30
```

**Run 2 (CUDA 1) — lr=2.5e-4 + T_max=30:**
```bash
CUDA_VISIBLE_DEVICES=1 python train.py \
  --dataset drivaerml \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --optimizer adamw \
  --lr 2.5e-4 \
  --weight-decay 1e-2 \
  --grad-clip 1.0 \
  --cosine-t-max 30 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb-group drivaerml-lr-sweep \
  --wandb-name driv-lr25e4-Tmax30
```

**Run 3 (CUDA 2) — lr=3e-4 + T_max=30:**
```bash
CUDA_VISIBLE_DEVICES=2 python train.py \
  --dataset drivaerml \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --optimizer adamw \
  --lr 3e-4 \
  --weight-decay 1e-2 \
  --grad-clip 1.0 \
  --cosine-t-max 30 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb-group drivaerml-lr-sweep \
  --wandb-name driv-lr3e4-Tmax30
```

## Baseline
- Dataset: DrivAerML
- Primary metric: percentage error (lower is better)
- Current best: **4.619%** (PR #2691, 4L/512d/8H + AdamW lr=5e-4 + T_max=30)

## Expected Outcome
Report percentage error for each run. If lr=3e-4 beats 4.619%, we have headroom to lower the LR further. The trend across lr ∈ {2e-4, 2.5e-4, 3e-4} will indicate whether there's a clear LR optimum below 5e-4 or if 5e-4 is already near-optimal for this architecture/dataset combo.

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing this PR because the assigned student pod was retired during the DrivAerML and cross-dataset refocus on 2026-04-21. We are freeing capacity for the new relaunch wave, and this lane is no longer being actively pursued. Any useful results already logged here or in W&B remain part of the research record.
